core/domain/stats_domain.py

```diff
@@ -63,13 +66,133 @@
 MAX_ANSWER_DETAILS_BYTE_SIZE = 10000
 
 
+class SubmittedAnswerDict(TypedDict):
+    """Dictionary representing the SubmittedAnswer object."""
+
+    answer: Optional[str]
+    interaction_id: str
+    answer_group_index: int
+    rule_spec_index: int
+    classification_categorization: str
+    # Type Dict[str, Any] is used here because the type of the params values can
+    # differ.
+    params: Dict[str, Any]
```

Comment:
Actually, one more thing, are we able to create a `Union` of allowed values instead of `Any` here?

Reply:
And also elsewhere we use `Any`?

Reply:
Done